src/core_init_atmosphere/mpas_init_atm_gwd.F

```diff
@@ -314,14 +334,17 @@ function read_global_30s_topo(path, sub_path) result(iErr)
       integer, parameter :: tile_y = 1200       ! y-dimension of each tile of global 30-arc-second topography
       integer, parameter :: tile_bdr = 3        ! number of layers of border/halo points surrounding each tile
 
-      integer :: istatus
+      integer (c_int) :: istatus
       integer :: ix, iy, ishift, ix_shift
-      integer :: isigned, endian, wordsize, nx, ny, nz
-      real (kind=R4KIND) :: scalefactor
-      real (kind=R4KIND), dimension(:,:,:), allocatable :: tile
+      integer (c_int) :: isigned, endian, wordsize, nx, ny, nz
+      real (c_float) :: scalefactor
+      real (c_float), dimension(:,:,:), pointer, contiguous :: tile
+      type (c_ptr) :: tile_ptr
       character(len=StrKIND) :: filename
+      character(len=StrKIND+1, kind=c_char) :: c_filename
```

Comment:
A `character(len=StrKIND+1, kind=c_char)` may be interoperable with a `character(kind=c_char), dimension(StrKIND+1)`, but since the latter is closer to what is used in the interface for `read_geogrid`, I'd propose we change the declarations of `c_filename` to be character arrays.

Reply:
Good catch. I updated this deceleration and the deceleration to `c_fname` within `mpas_init_atm_static.F`.